### PR TITLE
fix double reward claiming bug

### PR DESCRIPTION
**Pull request by emmazzz**
This PR fixes a bug reported in issue #3609 and adds two test cases for it.

### CODE REVIEW

File: crates/sui-framework/sources/governance/delegation.move [hunk 1]
```diff
@@ -191,7 +191,7 @@ module sui::delegation {
             self.next_reward_unclaimed_epoch <= epoch_to_claim
         } else {
             let ending_epoch = *option::borrow(&self.ending_epoch);
-            ending_epoch > epoch_to_claim
+            ending_epoch > epoch_to_claim && self.next_reward_unclaimed_epoch <= epoch_to_claim
```
Comment (lxfind) [inline crates/sui-framework/sources/governance/delegation.move:194]: somehow dedup with line 191?

Comment (emmazzz) [reply crates/sui-framework/sources/governance/delegation.move]: Gave it a try. Hopefully it's cleaner now.

**Review by lxfind — APPROVED** [review]

**Comment by lxfind** [comment]
Thank you for fixing this!

**Comment by emmazzz** [comment]
This PR is [failing CI](https://github.com/MystenLabs/sui/runs/7661423528?check_suite_focus=true#step:6:2757) at a testcase on gas: https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/unit_tests/gas_tests.rs#L436
The computation cost is off by one somehow. Any idea why the innocent changes here make this test fail? @lxfind

**Pull request merged**